### ISSUE TITLE
nc::percentile works only with nc::NdArray<double> array

### ISSUE THREAD

**Issue by ArMouReR** [dpilger26/NumCpp#98]
First of all thank you for the wonderful library and sorry for potentially dumb question.
I have face the following issue while trying to use nc::percentile:

The following example compiles and works without any issues:
```
nc::NdArray<double> test = { {3, 4, 1}, {3, 4, 1}, {6, 7, 8} };
auto low_dy = nc::percentile(test.ravel(), 99);
```

But if I change the type to float or int or nc::uint8 as in example below:
```
nc::NdArray<float> test = { {3, 4, 1}, {3, 4, 1}, {6, 7, 8} };
auto low_dy = nc::percentile(test.ravel(), 99);
```
I am failing with compilation errors of the following type:
`Error	C2398	Element '1': conversion from 'const double' to '_Tp' requires a narrowing conversion	...\NumCpp\Functions\percentile.hpp	115	`

```
if (inInterpMethod == "linear")
{
    const double percentI = static_cast<double>(indexLower) / static_cast<double>(inArray.size() - 1);
    const double fraction = (inPercentile / 100.0 - percentI) /
	(static_cast<double>(indexLower + 1) / static_cast<double>(inArray.size() - 1) - percentI);

    const double returnValue = arrayCopy[indexLower] + (arrayCopy[indexLower + 1] - arrayCopy[indexLower]) * fraction;
    NdArray<dtype> returnArray = { returnValue };
    return returnArray;
}
```
While the error is pointing to `NdArray<dtype> returnArray = { returnValue };` line.

About the environment: I am using VS 2019 with C++17 enabled.

What I have missed ?

**Comment by dpilger26** [comment]
Fixed in new release version 2.4.2

**Comment by ArMouReR** [comment]
Thanks a lot for so quick fix !
